x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/topn/TopNOperatorTests.java

```diff
@@ -1299,13 +1349,23 @@ public void testCloseWithoutCompleting() {
         ) {
             op.addInput(new Page(new IntArrayVector(new int[] { 1 }, 1).asBlock()));
         }
-        assertThat(breaker.getUsed(), equalTo(0L));
     }
 
+    private final List<CircuitBreaker> breakers = new ArrayList<>();
+
     @Override
     protected DriverContext driverContext() { // TODO remove this when the parent uses a breaking block factory
         BigArrays bigArrays = new MockBigArrays(PageCacheRecycler.NON_RECYCLING_INSTANCE, ByteSizeValue.ofGb(1)).withCircuitBreaking();
-        return new DriverContext(bigArrays, new BlockFactory(bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST), bigArrays));
+        CircuitBreaker breaker = bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST);
+        breakers.add(breaker);
+        return new DriverContext(bigArrays, new BlockFactory(breaker, bigArrays));
+    }
+
+    @After
+    public void allBreakersEmpty() {
+        for (CircuitBreaker breaker : breakers) {
+            assertThat(breaker.getUsed(), equalTo(0L));
```

Comment:
This is going to be useful. 👍

Reply:
Yeah. I think we should drag this into `AnyOperatorTests` - maybe pretty soon.